### PR TITLE
Update function signatures for `django.db.models.sql.Query` class

### PR DESCRIPTION
**Pull request by DongwookKim0823**
# I have made things!

This PR updates the function signatures for the `Query` class in `django.db.models.sql.query`. Additionally, it removes unnecessary functions.

### CODE REVIEW

File: django-stubs/db/models/sql/query.pyi [hunk 1]
```diff
@@ -81,51 +81,71 @@ class Query(BaseExpression):
...
-        branch_negated: bool = ...,
-        current_negated: bool = ...,
-        can_reuse: set[str] | None = ...,
-        allow_joins: bool = ...,
-        split_subq: bool = ...,
-        check_filterable: bool = ...,
-        update_join_types: bool = ...,
-    ) -> tuple[WhereNode, Iterable[str]]: ...
-    def add_filter(self, filter_clause: tuple[str, Any]) -> None: ...
+        filter_expr: Any,
```
Comment (sobolevn) [inline django-stubs/db/models/sql/query.pyi]: Why did you change `filter_expr`'s type?

File: django-stubs/db/models/sql/query.pyi [hunk 2]
```diff
@@ -81,51 +81,71 @@ class Query(BaseExpression):
...
+        branch_negated: bool = False,
+        current_negated: bool = False,
+        can_reuse: Any | None = None,
```
Comment (sobolevn) [inline django-stubs/db/models/sql/query.pyi]: And `can_reuse`?

File: django-stubs/db/models/sql/query.pyi (hunk 1 above)
Comment (DongwookKim0823) [reply django-stubs/db/models/sql/query.pyi]: The change was a mistake. I've reverted the type of filter_expr to its original state.

Comment (DongwookKim0823) [reply django-stubs/db/models/sql/query.pyi]: The change was a mistake. I've reverted the type to its original state.

File: django-stubs/db/models/sql/query.pyi [hunk 3]
```diff
@@ -81,51 +81,73 @@ class Query(BaseExpression):
...
     @cached_property
     def base_table(self) -> str: ...
+    def add_annotation(self, annotation: Any, alias: str, select: bool = True) -> None: ...
     def sql_with_params(self) -> tuple[str, tuple]: ...
     def __deepcopy__(self, memo: dict[int, Any]) -> Query: ...
-    def get_compiler(self, using: str | None = ..., connection: BaseDatabaseWrapper | None = ...) -> SQLCompiler: ...
+    def get_compiler(
+        self, using: str | None = None, connection: Any | None = None, elide_empty: bool = True
+    ) -> Any: ...
+    def join_parent_model(self, opts: Any, model: Any | None, alias: str, seen: dict[Any | None, str]) -> str: ...
```
Comment (sobolevn) [inline django-stubs/db/models/sql/query.pyi]:
```suggestion
    def join_parent_model(self, opts: Any, model: Any | None, alias: str, seen: dict[Any, str]) -> str: ...
```

**Review by sobolevn — APPROVED** [review]
Thank you!
